lib/chewy/index/actions.rb

```diff
@@ -163,7 +163,29 @@ def #{method} options = {}
         def reset!(suffix = nil, journal: false)
           if suffix.present? && (indexes = self.indexes).present?
             create! suffix, alias: false
-            result = import suffix: suffix, journal: journal
+
+            result =
+              if Chewy.use_enhance_index_settings_while_resetting
+                name = build_index_name(suffix: suffix)
+                client.indices.put_settings index: name, body: { index: { number_of_replicas: 0, refresh_interval: -1 } }
```

Comment:
Can we refactor this? Make `number_of_replicas` a separate method with the block or so. I mean, this method became way huge, right?